### ISSUE TITLE
Loading a tensor from device data pointer.

### ISSUE THREAD

**Issue by omerbenamram** [LaurentMazare/tch-rs#265]
Hi there!

I'm trying to create a `tch::Tensor` from existing inference output data on a CUDA device.
I was trying to see if there is the API described in [https://discuss.pytorch.org/t/create-tensor-from-data-pointer/19708](https://discuss.pytorch.org/t/create-tensor-from-data-pointer/19708) exists, but `tch` doesn't seem to have the notion of context objects, not the `fromBlob` api.

If I have a data pointer on the host, this indeed seems to work:
```
        Tensor::of_data_size(
            std::slice::from_raw_parts(
                self.data_ptr,
                (self.number_of_elements * 2) as usize, // *2 because of fp16
            ),
            &self.dims,
            tch::Kind::Half,
        )
```

But I haven't found any API that also accepts this on a CUDA device.

Is this something that makes sense considering the architecture of `tch`?

Thanks alot for this wonderful crate :)

**Comment by omerbenamram** [comment]
I did end up implementing the wrapper around `torch::from_blob`, and this works, however as per the docs, `tch` will raise an exception if we try to manipulate the memory afterwards, but it did solve my case of a read only view into an existing CUDA mapped data.

https://github.com/LaurentMazare/tch-rs/compare/master...omerbenamram:feature/tensor_of_blob

**Comment by omerbenamram** [comment]
closed by #273.
